source/find_getmore_killcursors_commands.rst

```diff
@@ -365,18 +365,38 @@ More detailed information about the interaction of the **secondaryOk** with **OP
 Behavior of Limit, skip and batchSize
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The new **find** command has different semantics to the existing 3.0 and earlier **OP_QUERY** wire protocol message. The **limit** field is a hard limit on the total number of documents returned by the cursor no matter what **batchSize** is provided.
-
-Once the limit on the cursor has been reached the server will destroy the cursor and return a **cursorId** of **0** in the **OP_REPLY**. This differs from existing **OP_QUERY** behavior where there is no server side concept of limit and where the driver **MUST** keep track of the limit on the client side and **MUST** send a **OP_KILL_CURSORS** wire protocol message when it limit is reached.
-
-When setting the **batchSize** on the **find** and **getMore** command the value MUST be based on the cursor limit calculations specified in the `CRUD`_ specification. 
-
-In the following example the **limit** is set to **4** and the **batchSize** is set to **3** the following commands are executed.
+The new **find** command has different semantics to the existing 3.0 and earlier
+**OP_QUERY** wire protocol message. The **limit** field is a hard limit on the
+total number of documents returned by the cursor no matter what **batchSize** is
+provided. This includes other limiting operations, such as the **$limit**
+aggregation pipeline stage. This differs from existing **OP_QUERY** behavior
+where there is no server-side concept of limit and where the driver **MUST**
+keep track of the limit on the client side and **MUST** send a
+**OP_KILL_CURSORS** wire protocol message when the limit is reached.
+
+When setting the **batchSize** on the **find** and **getMore** commands the
+value of **batchSize** **MUST** be based on the cursor limit calculations
+specified in the `CRUD`_ specification.
+
+Note that with 5.0, the server-side handling of cursors with a limit has
+changed. Before 5.0, some cursors were automatically closed when the limit was
+reached (e.g. when running **find** with **limit**), and the reply document did
+not include a cursor ID (i.e. ``cursor.id`` was ``0``). Starting with 5.0, all
+cursor-producing operations will return a cursor ID if the end of the batch
+being returned lines up with the limit on the cursor. In this case, drivers
+**MUST** ensure the cursor is closed on the server, either by exhausting the
+cursor or by using **killCursors** to kill it.
+
+In the following example the **limit** is set to **4** and the **batchSize** is
+set to **3** the following commands are executed. The last command is either
+**killCursors** or **getMore**, depending on how a driver ensures the cursor is
+closed on 5.0:
 
 .. code:: javascript
 
-    {find: ..., batchSize:3}
-    {getMore: ..., batchSize:1}
+    {find: ..., batchSize:3, limit:4}
+    {getMore: ..., batchSize:1} // Returns remaining items but leaves cursor open on 5.0
+    {...}          // Kills server-side cursor. Necessary on
```

Comment:
I think there's some missing text here: "Necessary on 5.0+?"

Reply:
Yes. Thank you